### ISSUE TITLE
False positive with MagicNumber on default value for parameters

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#1055]
Correct me if I'm wrong but the following method signature gets flagged with a `MagicNumber` with Detekt 1.0.0.RC8:

```kotlin
fun Context.vibrate(duration: Long = 300L)
```

I thought that default parameters won't be treated as such code smells. As for the configuration I'm using:

```yml
failFast: true

potential-bugs:
  UnsafeCast:
    active: false # We know what we're doing.
  LateinitUsage:
    active: false # We know what we're doing.

naming:
  FunctionMinLength:
    active: false
  FunctionMaxLength:
    active: false

style:
  TopLevelPropertyNaming:
    active: false
  MaxLineLength:
    active: false
  UnnecessaryParentheses:
    active: false # False positives.

complexity:
  LongParameterList:
    threshold: 8

comments:
  UndocumentedPublicClass:
    active: false
  UndocumentedPublicFunction:
    active: false
```

**Comment by schalkms** [comment]
Detekt hasn't considered this case yet. `MagicNumber` allows constant declarations in constructors, but as it looks like not in functions.
We could need some help here. Would you mind taking a look at this @vanniktech ?

**Comment by vanniktech** [comment]
Eventually yeah but don't get your hopes up.
